feat: support changing file modes

ribru17 opened the pull request:
This commit also properly exposes the similarity index as a node (previously `field` would not work for anonymous regex nodes). It also updates the Tree-sitter version, which generated the associated new bindings.

EDIT: It also exposes the argument portion of the `(command)` node

the-mikedavis commented:
You're saying that `(similarity score: _ @somecapture)` doesn't work when that field is just a regex? Switching to a named `score` node instead seems ok to me.

Could you undo the unrelated changes? The PR can just be the line about the `similarity` node - the CI will take care of regenerating the `parser.c`

ribru17 commented:
> Could you undo the unrelated changes? The PR can just be the line about the `similarity` node - the CI will take care of regenerating the `parser.c`

Sure, sorry for the giant diff :laughing: 

> You're saying that `(similarity score: _ @somecapture)` doesn't work when that field is just a regex? Switching to a named `score` node instead seems ok to me.

For me, no: I got invalid structure complaints. Also I was never able to even see the field exposed in the parsed tree

ribru17 committed:
Expose diff argument as a named 'argument' node

ribru17 committed:
Expose the similarity score as a named node

It was previously a field, but a field is an impossible pattern for a
regex literal like we use for score. I.e.:

    (similarity score: _ @_some_capture_)

Cannot match and on recent tree-sitter versions will cause an error
during query analysis.

ribru17 committed:
Support mode changing syntax

Changes that are only permissions take the shape `old/new mode <mode>`.

the-mikedavis commented:
Thanks! I've pushed to split the changes up a little, so the commit history is easier to follow.

ribru17 commented:
Of course, and thank you!
